rohmu/object_storage/s3.py

```diff
@@ -24,16 +24,50 @@
 )
 from botocore.response import StreamingBody
 from rohmu.util import batched
-from typing import Any, BinaryIO, Collection, Iterator, Optional, Tuple, Union
+from typing import Any, BinaryIO, cast, Collection, Iterator, Optional, Tuple, TYPE_CHECKING, Union
 
-import boto3
 import botocore.client
 import botocore.config
 import botocore.exceptions
 import botocore.session
 import math
 import time
 
+# botocore typing stubs are incomplete. We either have to write all the stubs we need
+# or work around the problem a bit by importing the typing library only for type checking purposes.
+if TYPE_CHECKING:
+    from mypy_boto3_s3 import S3Client
+
+    s3_client_type = S3Client
+else:
+    s3_client_type = Any
```

Comment:
I feel a bit dirty for doing this, but it does seem to have the expected effect

Reply:
I don't think it is needed, as with future annotation import, the annotations are not evaluated anyway but instead are strings. Due to that, in non TYPE_CHECKING they should not be used.